apps/emqx_authz/src/emqx_authz.erl

```diff
@@ -364,8 +363,95 @@ type(<<"postgresql">>) -> postgresql;
 type('built_in_database') -> 'built_in_database';
 type(<<"built_in_database">>) -> 'built_in_database';
 %% should never happen if the input is type-checked by hocon schema
-type(Unknown) -> error({unknown_authz_source_type, Unknown}).
+type(Unknown) -> throw({unknown_authz_source_type, Unknown}).
+
+
+maybe_write_files(Source) ->
+    maybe_write_files(Source, #{}).
+
+maybe_write_files(#{<<"type">> := <<"file">>} = Source, _OldSource) ->
+    write_acl_file(Source);
+maybe_write_files(NewSource, #{}) ->
+    maybe_write_certs(NewSource);
+maybe_write_files(NewSource, OldSource) ->
+    maybe_write_certs(NewSource, OldSource).
+
+write_acl_file(#{<<"rules">> := Rules,
+                 <<"enable">> := Enable} = Source)
+  when ?IS_ENABLED(Enable) ->
+    %% rules from api, maybe path.
+    %% Otherwise will be placed in `authz` subdirectory inside EMQX's `data_dir`
+    %% Maybe failed with no write access on remote nodes.
+    Path = maps:get(<<"path">>, Source, acl_conf_file()),
+    {ok, _Filename} = write_file(Path, Rules),
+    Source#{<<"path">> => Path};
+write_acl_file(#{<<"path">> := Path,
+                 <<"enable">> := Enable} = Source)
+  when ?IS_ENABLED(Enable) ->
+    %% Only path from API, file content will be synced by EMQX.
+    %% Maybe failed with no read(local node) or write(remote) access.
+    try file:read_file(Path) of
+        {ok, Rules} ->
+            Source#{<<"rules">> => Rules}
+    catch
+        _ : _ ->
```

Comment:
This doesn’t catch error returns from file:read_file.
Use “case of” instead of “try of”